test/e2e-secondary-network/secondary_network_test.go

```diff
@@ -116,34 +166,52 @@ func (data *testData) createPodForSecondaryNetwork(ns string, pod *testPodInfo)
 	return podBuilder.Create(data.e2eTestData)
 }
 
-// listPodIPs returns a map of Pod IPs, indexed by the interface name. All interfaces are included
-// and only IPv4 addresses are considered. If an interface is not assigned an IPv4 address, it will
-// be included in the map, with a nil value.
-func (data *testData) listPodIPs(targetPod *testPodInfo) (map[string]net.IP, error) {
+// listPodAddresses returns Pod network interface information with enhanced parsing.
+func (data *testData) listPodAddresses(targetPod *testPodInfo) (map[string]net.IP, map[string]net.IP, map[string]string, error) {
 	cmd := []string{"ip", "addr", "show"}
 	stdout, _, err := data.e2eTestData.RunCommandFromPod(data.e2eTestData.GetTestNamespace(), targetPod.podName, containerName, cmd)
 	if err != nil {
-		return nil, fmt.Errorf("error when listing interfaces for %s: %w", targetPod.podName, err)
+		return nil, nil, nil, fmt.Errorf("error when listing interfaces for %s: %w", targetPod.podName, err)
 	}
-	result := make(map[string]net.IP)
+	ipv4Result := make(map[string]net.IP)
+	macResult := make(map[string]string)
+	ipv6Result := make(map[string]net.IP)
 	var currentInterface string
 	lines := strings.Split(stdout, "\n")
+	inet6Num := make(map[string]int)
 	for _, line := range lines {
 		fields := strings.Fields(line)
 		if len(fields) >= 2 && strings.HasSuffix(fields[0], ":") {
 			// first field is ifindex, second field is interface name
 			currentInterface = strings.Split(strings.TrimSuffix(fields[1], ":"), "@")[0]
-			result[currentInterface] = nil
-		} else if len(fields) >= 2 && fields[0] == "inet" {
+			ipv4Result[currentInterface] = nil
+			ipv6Result[currentInterface] = nil
+			macResult[currentInterface] = ""
+			inet6Num[currentInterface] = 0
+		} else if len(fields) >= 2 && (fields[0] == "inet" || fields[0] == "inet6") {
 			ipStr := strings.Split(fields[1], "/")[0]
 			ip := net.ParseIP(ipStr)
 			if ip == nil {
-				return nil, fmt.Errorf("failed to parse IP (%s) for interface %s of Pod %s", ipStr, currentInterface, targetPod.podName)
+				return nil, nil, nil, fmt.Errorf("failed to parse IP (%s) for interface %s of Pod %s", ipStr, currentInterface, targetPod.podName)
 			}
-			result[currentInterface] = ip
+			if fields[0] == "inet" {
+				ipv4Result[currentInterface] = ip
+			} else if fields[0] == "inet6" {
+				inet6Num[currentInterface] += 1
+				if ipv6Result[currentInterface] == nil {
+					ipv6Result[currentInterface] = ip
+				}
+			}
```

Comment:
```suggestion
			}
			if fields[0] == "inet6" {
				inet6Num[currentInterface] += 1
				if ipv6Result[currentInterface] == nil {
					ipv6Result[currentInterface] = ip
				}
			}
```
It seems here you are assuming the global IPv6 is always before the local link IPv6 and skip assigning the second IPv6 address to the current interface. It's probably not true. Can you double check?

Reply:
https://github.com/antrea-io/antrea/pull/7069#discussion_r2085079194
Good catch, and considering this discussion, I think we can check the `scope global` keyword and only use the global IPv6 in the `ipv6Result` map.